Add `align-temporal-unit-with-param-type-and-value` multimethod (#37813)
metabase/metabase: docs/developers-guide/driver-changelog.md

  an additional argument: `database`.
  The new function arglist is `[driver database connection schema-inclusion-filters schema-exclusion-filters]`.

- The method `metabase.driver.sql.parameters.substitution/align-temporal-unit-with-param-type` is now deprecated.
  Use `metabase.driver.sql.parameters.substitution/align-temporal-unit-with-param-type-and-value` instead,

>   which has the access to `value` and intended to provide more flexibility on choosing the right conversion unit.

Reviewer: ```suggestion
  which has access to `value` and therefore provides more flexibility for choosing the right conversion unit.
```
